src/server/pkg/storage/fileset/index/writer.go

```diff
@@ -0,0 +1,159 @@
+package index
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/chunk"
+)
+
+const (
+	indexType        = 'i'
+	rangeType        = 'r'
+	defaultRangeSize = int64(chunk.MB)
```

Comment:
I think it would be good to have a comment here (or maybe on `NewWriter` below) describing what range size corresponds to and what effects changing it will have. Also how important do you think it is to have it configurable in `NewWriter` rather than just a global constant. Are we going to want to have different writers with different range sizes?

Reply:
The only reason I have it configurable is for testing purposes. Setting it low allows me to create multiple index levels with a small amount of data in my tests. In a real deployment, the default range size would be used. I can add a comment that it should only be set for testing purposes.